### DISCUSSION TITLE
Api Key: Add encrypt/decript the key

### DISCUSSION THREAD

**Discussion by biapar** [marcominerva/SimpleAuthentication#107] [Ideas]
Api Key: Add encrypt/decript the key server side to protect the real api key. The client pass the key ( encrypted ). The server decrypt the key and then follow the rest.

**Comment by marcominerva** [comment]
How should client and server share encryption information?

**Reply by biapar** [reply]
Have you any idea? I wish to use the apikey into a vuejs client.

**Reply by marcominerva** [reply]
Api Keys are themselves a kind of "secret", a lot of services uses them as they are (OpenAI, Azure Services, etc.). What is the benefit of encrypting in these scenarios?

**Reply by biapar** [reply]
I think that a "bad" person can take the ApiKey by the browser developer tools or from the js code ( client side ).

**Reply by marcominerva** [reply]
OK, but he can take it also if it is encrypted...

**Reply by biapar** [reply]
Uhm..How to?

**Reply by marcominerva** [reply]
Opening the Browser Developer Console and inspecting the network traffic. It is just an header.

**Reply by biapar** [reply]
Yes, I know...but I don't understand how to complicate the life of a "bad person" using api-key.

**Reply by marcominerva** [reply]
You can for example use CORS to restrict what browser application can access the server.

For more information, take a look to this post: https://stackoverflow.com/a/59526306/1728189.

**Reply by biapar** [reply]
Thanks...I'll try to add CORS with "origin: https://www.sito.it" plus apikey. I'll inform you. PS: Do you maintain this repo?
